cmake/toolchains/ARM.cmake

```diff
@@ -34,7 +36,7 @@ target_compile_options(mbed-os
 
 target_compile_options(mbed-os
     PUBLIC
-        $<$<COMPILE_LANGUAGE:ASM>:${MBED_STUDIO_ARM_COMPILER}>
+        $<$<COMPILE_LANGUAGE:ASM>:${mbed_studio_arm_compiler}>
```

Comment:
why was this changed to lower case?

Reply:
Because the global property `MBED_STUDIO_ARM_COMPILER` is accessed as `mbed_studio_arm_compiler` in this module. It was changed to a global property so it can also be used in the one of the new functions.